sdl_ios/SmartDeviceLink/SDLProtocol.m

```diff
@@ -237,22 +253,52 @@ - (void)processMessages {
 }
 
 - (void)sendHeartbeat {
-	SDLProtocolHeader *header = [SDLProtocolHeader headerForVersion:self.version];
+    SDLProtocolHeader *header = [SDLProtocolHeader headerForVersion:self.version];
     header.frameType = SDLFrameType_Control;
     header.serviceType = 0;
     header.frameData = SDLFrameData_Heartbeat;
-    header.sessionID = self.sessionID;
-
+    
     SDLProtocolMessage *message = [SDLProtocolMessage messageWithHeader:header andPayload:nil];
 
     [self sendDataToTransport:message.data withPriority:header.serviceType];
+    
+}
 
+- (void)sendRawDataStream:(NSInputStream *)inputStream withServiceType:(SDLServiceType)serviceType {
+    
+    SDLDataStreamHandlingDelegate *streamDelegate = [SDLDataStreamHandlingDelegate new];
+    streamDelegate.serviceType = serviceType;
+    streamDelegate.protocol = self;
+    objc_setAssociatedObject(inputStream, @"RetainedDelegate", streamDelegate, OBJC_ASSOCIATION_RETAIN);
```

Comment:
If an associated object _is_ to be used, it should be a category on NSInputStream and the key should be a selector of the getter in order to guarantee its uniqueness. See the [NSHipster article of Associated Objects](http://nshipster.com/associated-objects/) for more information. Also please note the last two paragraphs..

>Associated objects should be seen as a method of last resort, rather than a solution in search of a problem (and really, categories themselves really shouldn't be at the top of the toolchain to begin with).

>Like any clever trick, hack, or workaround, there is a natural tendency for one to actively seek out occasions to use it—especially just after learning about it. Do your best to understand and appreciate when it's the right solution, and save yourself the embarrassment of being scornfully asked "why in the name of $DEITY" you decided to go with that solution.